### DISCUSSION TITLE
Extend basic startup permissions

### DISCUSSION THREAD

**Discussion by devopg** [telegraf/telegraf#1780] [Q&A] [answered]
Hello. How can i get and extend default startup permissions with myself? 
like:

```
var permission = Telegraf.GetDefaultLaunchPermission();
var myPermission =
{
    allowedUpdates: ['chat_member']
}
permission.allowedUpdates.push(myPermission.allowedUpdates[0]);

bot.launch(permission);
```

**Comment by hypernova7** [comment]
You can't. allowed_updates is generally set for all chat types. Therefore it cannot be configured individually

**Reply by devopg** [reply]
where can i find default setting then start bot like:?
`bot.launch();`

Now problem: then i start with:
`bot.launch({ allowedUpdates: ['chat_member']  });`
dont work:
```
bot.on(message('sticker'
bot.command('mycommand'
bot.hears("mytext"
```

**Reply by MKRhere** [reply]
All updates:

```TS
bot.launch({
  allowedUpdates: [
    "message",
    "edited_message",
    "channel_post",
    "edited_channel_post",
    "inline_query",
    "chosen_inline_result",
    "callback_query",
    "shipping_query",
    "pre_checkout_query",
    "poll",
    "poll_answer",
    "my_chat_member",
    "chat_member",
    "chat_join_request",
  ],
});
```

Pick the ones relevant to you.

This list is from: https://core.telegram.org/bots/api#getting-updates
All fields except `update_id` are update types.

**Comment by MKRhere** [comment, marked as answer]
There's no default list because of this:

---

From https://core.telegram.org/bots/api#getupdates

### allowed_updates

A JSON-serialized list of the update types you want your bot to receive. For example, specify `[“message”, “edited_channel_post”,  “callback_query”]` to only receive updates of these types. See Update for a complete list of available update types. Specify an empty list to receive all update types except `chat_member` (default).

---

By passing empty, it defaults to everything except `chat_member`.

If you use an editor that supports TypeScript, even if you write JS, you'll get Intellisense to help you with this. If you use VSCode, ctrl+space will show the full list:

![image](https://user-images.githubusercontent.com/19621208/212034065-f9dcd062-e493-4327-be22-aab7df79b03a.png)

For your immediate reference, reposting the full list here:

```TS
bot.launch({
  allowedUpdates: [
    "message",
    "edited_message",
    "channel_post",
    "edited_channel_post",
    "inline_query",
    "chosen_inline_result",
    "callback_query",
    "shipping_query",
    "pre_checkout_query",
    "poll",
    "poll_answer",
    "my_chat_member",
    "chat_member",
    "chat_join_request",
  ],
});
```

**Comment by devopg** [comment]
Thanks, I got it.
A little theoretical question, why only "chat_member" was excluded? :thinking:

**Reply by MKRhere** [reply]
You'll have to ask the Telegram team.
